be/test/vec/data_types/data_type_array_test.cpp

```diff
@@ -0,0 +1,490 @@
+    DataTypeSerDeSPtrs serdes;
+};
+
+// ================ common data type and serder test =================
+TEST_F(DataTypeArrayTest, MetaInfoTest) {
+    for (int i = 0; i < array_types.size(); i++) {
+        auto& type = array_types[i];
+        auto& desc = array_descs[i];
+        auto nested_type =
+                assert_cast<const DataTypeArray*>(remove_nullable(type).get())->get_nested_type();
+
+        TypeDescriptor arr_type_descriptor = {PrimitiveType::TYPE_ARRAY};
+        arr_type_descriptor.add_sub_type(desc[0].type_desc.children[0]);
+        auto col_meta = std::make_shared<PColumnMeta>();
+        Array a;
+        a.push_back(nested_type->get_default());
+        col_meta->set_type(PGenericType_TypeId_LIST);
+        DataTypeMetaInfo arr_meta_info_to_assert = {
+                .type_id = TypeIndex::Array,
+                .type_as_type_descriptor = &arr_type_descriptor,
+                .family_name = "Array",
+                .has_subtypes = true,
+                .storage_field_type = doris::FieldType::OLAP_FIELD_TYPE_ARRAY,
+                .should_align_right_in_pretty_formats = false,
+                .text_can_contain_only_valid_utf8 = nested_type->text_can_contain_only_valid_utf8(),
+                .have_maximum_size_of_value = false,
+                .size_of_value_in_memory = size_t(-1),
+                .precision = size_t(-1),
+                .scale = size_t(-1),
+                .is_null_literal = false,
+                .is_value_represented_by_number = false,
+                .pColumnMeta = col_meta.get(),
+                .default_field = a,
+        };
+        DataTypePtr arr = remove_nullable(type);
+        meta_info_assert(arr, arr_meta_info_to_assert);
+    }
+}
+
+TEST_F(DataTypeArrayTest, CreateColumnTest) {
+    // for tinyint -> largeint, float,double, ipv4/6, all date type  here uncompressed size is 16
+    // for string here uncompressed size is 24
+    for (int i = 0; i < 13; i++) {
+        auto type = remove_nullable(array_types[i]);
+        // any different nested type in arr with same default array ?
+        Field default_field_array = Array();
+        create_column_assert(type, default_field_array, 16);
+    }
+    {
+        auto type = remove_nullable(array_types[13]);
+        Field default_field_array = Array();
+        create_column_assert(type, default_field_array, 24);
+    }
+    // for decimal32/64/128/256 here uncompressed size is 16
+    for (int i = 14; i < array_types.size(); i++) {
+        auto type = remove_nullable(array_types[i]);
+        Field default_field_array = Array();
+        create_column_assert(type, default_field_array, 16);
+    }
+}
+
+TEST_F(DataTypeArrayTest, GetFieldTest) {
+    TExprNode node;
+    node.node_type = TExprNodeType::ARRAY_LITERAL;
+    for (size_t i = 0; i < array_columns.size(); ++i) {
+        Field assert_field;
+        array_columns[i]->get(0, assert_field);
+        get_field_assert(array_types[i], node, assert_field, true);
```

Comment:
What do you check since there is no data in node and assert_false = true?

Reply:
just check this interface should throw exception